### ISSUE TITLE
Non-deterministic Results Despite Fixing Random Seeds  in video summarization

### ISSUE THREAD

**Issue by Doran-fans** [IntelLabs/GraVi-T#7]
Hi, thank you for sharing the code for VideoSAGE — really appreciate your contribution!

I noticed that the results vary across different runs, even when I fix all random seeds. I added the following function to control randomness:

def seed_torch(seed=1029):
    import os, random, numpy as np
    import torch

    random.seed(seed)
    os.environ['PYTHONHASHSEED'] = str(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)

    torch.backends.cudnn.benchmark = False
    torch.backends.cudnn.deterministic = True
    torch.use_deterministic_algorithms(True)

Still, the results are not deterministic.
Just wondering — is this expected behavior? Or are there any additional steps needed to fully control randomness in your code?
Thanks again for your great work!

Best regards,
ruifanzhao

**Comment by Doran-fans** [comment]
examples of same code result:
Final average results: F1-Score = 48.8, Kendall's Tau = 0.133, Spearman's Rho = 0.176
Final average results: F1-Score = 42.94, Kendall's Tau = 0.118, Spearman's Rho = 0.159
Final average results: F1-Score = 42.38, Kendall's Tau = 0.116, Spearman's Rho = 0.158
Final average results: F1-Score = 43.76, Kendall's Tau = 0.129, Spearman's Rho = 0.175
Final average results: F1-Score = 44.13, Kendall's Tau = 0.0685, Spearman's Rho = 0.0921
